TestValidateAsset() in appimport_test.go needs to use unique dirs, clean up

## What happened (or feature request):

I noticed in a test run the error `2017/05/05 08:14:28 ERROR:mkdir testing/data: file exists` and found that in TestValidateAsset() we're creating directories right in the user homedir. 

## What you expected to happen:

* Normally directories would be created via tmpdir, but this is a unique case and what it's testing is valid.
* However, directory names should be unique so that we can't have a later collision, use a way to create a tmpdir in ~/tmp or something.
* Make sure artifacts are cleaned up (This may already be being done, maybe I hit ctrl-c)

## How to reproduce it (as minimally and precisely as possible):

```
mkdir -p pkg/appimport/testing/data
make testpkg
```

## Anything else do we need to know:


## Related source links or issues:

`mkdir ~/testpath && make testpkg` will show this. 

```
=== RUN   TestValidateAsset
--- FAIL: TestValidateAsset (0.00s)
        Error Trace:    appimport_test.go:44
	Error:		Received unexpected error mkdir /Users/rfay/testpath: file exists

FAIL
FAIL	github.com/drud/ddev/pkg/appimport	0.019s
```

It's not that big a deal, but we should probably find a way not to create things in dev's home dir.

--- a/pkg/appimport/appimport_test.go
+++ b/pkg/appimport/appimport_test.go
@@ -38,12 +38,13 @@ func TestValidateAsset(t *testing.T) {
 
 	// test tilde expansion
 	userDir, err := homedir.Dir()
-	testDir := filepath.Join(userDir, "testpath")
+	testDirName := "testpath-" + util.RandString(4)
+	testDir := filepath.Join(userDir, testDirName)
 	assert.NoError(err)
 	err = os.Mkdir(testDir, 0755)
 	assert.NoError(err)
 
-	testPath, err := ValidateAsset("~/testpath", "files")
+	testPath, err := ValidateAsset("~/"+testDirName, "files")
 	assert.NoError(err)
 	assert.Contains(testPath, userDir)
 	assert.False(strings.Contains(testPath, "~"))

--- a/pkg/appimport/appimport_test.go
+++ b/pkg/appimport/appimport_test.go
@@ -38,7 +38,7 @@ func TestValidateAsset(t *testing.T) {
 
 	// test tilde expansion
 	userDir, err := homedir.Dir()
-	testDirName := "testpath-" + util.RandString(4)
+	testDirName := "tmp.ddev.testpath-" + util.RandString(4)
 	testDir := filepath.Join(userDir, testDirName)
 	assert.NoError(err)
 	err = os.Mkdir(testDir, 0755)

